src/main/java/com/google/api/codegen/config/ProtoApiModel.java

```diff
@@ -114,14 +115,29 @@ public String getDocumentationSummary() {
   }
 
   @Override
-  public Iterable<ProtoTypeRef> getTypes() {
+  public Iterable<ProtoTypeRef> getAdditionalTypes() {
     ImmutableList.Builder<ProtoTypeRef> models = ImmutableList.builder();
-    for (TypeRef t : new InterfaceView().getTypes(protoModel)) {
+    for (TypeRef t : getTypes(protoModel)) {
       models.add(new ProtoTypeRef(t));
     }
     return models.build();
   }
 
+  /**
+   * Helper to extract the types from the underlying model.
+   *
+   * @see com.google.api.Service#getTypesList()
+   * @param model model with service config
+   * @return types
+   */
+  protected Iterable<TypeRef> getTypes(Model model) {
```

Comment:
Why not `private`? (That is, will this be needed in subclasses of `ProtoApiModel`?)

Reply:
Not really, I simply prefer protected per my own conventions. Do we have some standard conventions that would discourage this?

Reply:
The internal Java style guide would have this be private, but the external one seems to be more lenient. I'll follow up offline.